src/components/Microsoft.DotNet.UpgradeAssistant.Analysis/AnalyzeResultWriter.cs

```diff
@@ -32,10 +33,26 @@ public async Task WriteAsync(IAsyncEnumerable<AnalyzeResultDefinition> results,
             {
                 Runs = await ExtractRunsAsync(results).ToListAsync(token).ConfigureAwait(false),
             };
+            if (format != null && format.ToLower(CultureInfo.CurrentCulture) == "html")
+            {
+                WriteHTML(sarifLog);
+            }
+            else
+            {
+                _serializer.Write(_sarifLogPath, sarifLog);
+            }
+        }
 
-            _serializer.Write(_sarifLogPath, sarifLog);
+        private void WriteHTML(SarifLog sarifLog)
+        {
+            var template = File.ReadAllText("../../../../../HTMLTemplate.html");
```

Comment:
Have you considered using Razor templates for this?

Reply:
Oh - this should be an embedded stream or a file located next to the assembly. File path stuff like this will fail quickly once outside of a dev environment.

Reply:
Resolved